proposals/lip-0041.md

```diff
@@ -97,23 +103,216 @@ In this section, we describe the properties of an auth account and specify their
+
+```java
+multisigRegMsgSchema = {
+    "type": "object",
+    "required": ["address", "nonce", "numberOfSignatures", "mandatoryKeys",  "optionalKeys"],
+    "properties": {
+        "address": {
+            "dataType": "bytes",
+            "length" : 20
+            "fieldNumber": 1
+        },
+        "nonce": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        },
+        "numberOfSignatures": {
+            "dataType": "uint32",
+            "fieldNumber": 3
+        },
+        "mandatoryKeys": {
+            "type": "array",
+            "items": {
+                "dataType": "bytes",
+                "length" : 32
+            },
+            "fieldNumber": 4
+        },
+        "optionalKeys": {
+            "type": "array",
+            "items": {
+                "dataType": "bytes",
+                "length" : 32
+            },
+            "fieldNumber": 5
+        }
+    }
+}
+```
 
 ##### Verification
 
-The verification of this command is defined in [LIP 0017](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0017.md#multisignature-account-registration).
+These specifications supersede those defined in [LIP 0017](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0017.md#multisignature-account-registration).
+
+```python
+def verify(trs: Transaction) -> None:
+    # Check that a multisignature group is not already registered for this address
+    senderAddress = 20-byte address derived from trs.senderPublicKey
+    if len(authAccount(senderAddress).mandatoryKeys) > 0 or len(authAccount(senderAddress).mandatoryKeys) > 0:
+        raise Exception('Multisignature group already registered for this address.')
+    # Mandatory keys are pairwise distinct and ordered lexicographically.
+    if trs.params.mandatoryKeys != sorted(set(trs.params.mandatoryKeys)):
+        raise Exception('Invalid mandatory keys array.')
+    # Optional keys are pairwise distinct and ordered lexicographically.
+    if trs.params.optionalKeys != sorted(set(trs.params.optionalKeys)):
+        raise Exception('Invalid optional keys array.')
+    # Mandatory and optional keys must be disjoint.
+    all_keys = trs.params.mandatoryKeys + trs.params.optionalKeys
+    if len(all_keys) != len(set(all_keys)):
+        raise Exception('Mandatory and optional keys arrays are not disjoint.')
+    # Union of mandatory and optional keys must contain between 1 and 64 keys (included).
+    if len(all_keys) < 1 or len(all_keys) > 64:
+        raise Exception('Invalid length of mandatory and optional keys arrays.')
+    # Number of signatures must be an integer between 1 and the total number of public keys.
+    if trs.params.numberOfSignatures < 1 or trs.params.numberOfSignatures > len(all_keys):
+        raise Exception('Invalid number of signatures.')
```

Comment:
As we have also a `signatures` property in the transaction, this may be confusing. Consider:
- Changing the error to "Invalid numberOfSignatures property".

Reply:
changed several error messages